File: src/main/java/com/dontwaste/model/entity/ProductTemplate.java
package com.dontwaste.model.entity;

import com.dontwaste.model.base.BaseId;
import com.dontwaste.model.entity.product.productType.DishType;
import com.dontwaste.model.entity.product.productType.KitchenType;
import lombok.*;
import org.hibernate.validator.constraints.Length;

import javax.persistence.*;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Builder
@Entity
@Table(name = "PRODUCT_TEMPLATE")
public class ProductTemplate extends BaseId {

    @Column(name = "PRODUCT_NAME", length = 30)
    private String productName;

    @Column(name = "PRODUCT_DESCRIPTION")
    @Length(max = 500)
    private String productDescription;

    @Column(name = "PRODUCT_IMAGE", nullable = false)
    @Length(max = 2600)
    private String productImage;

    @Column(name = "KOSHER")
    private boolean kosher;

    @Column(name = "VEGETARIAN")
    private boolean vegetarian;

    @Column(name = "VEGAN")
    private boolean vegan;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "INSTITUTION_ID", referencedColumnName = "ID")
    private Institution institution;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "KITCHENTYPE_ID", referencedColumnName = "ID")
    private KitchenType kitchenType;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "DISHTYPE_ID", referencedColumnName = "ID")
    private DishType dishType;




}

File: Front/src/app/-Models-/product-template.ts
import {Institution} from "./institution";
import {KitchenType} from "./kitchen-type";
import {DishType} from "./dish-type";

export class ProductTemplate {
  id: number;
  productName: string;
  productDescription: string;
  productImage: string;
  institution: any;
  kitchenType: any;
  dishType: any;
  kosher: boolean;
  vegetarian: boolean;
  vegan: boolean;


  constructor(id: number,
              productName: string,
              productDescription: string,
              productImage: string,
              institution: Institution,
              kitchenType: KitchenType,
              dishType: DishType,
              kosher: boolean,
              vegetarian: boolean,
              vegan: boolean) {
    this.id = id;
    this.productName = productName;
    this.productDescription = productDescription;
    this.productImage = productImage;
    this.institution = institution;
    this.kitchenType = kitchenType;
    this.dishType = dishType;
    this.kosher = kosher;
    this.vegetarian = vegetarian;
    this.vegan =  vegan;
  }
}
